torch/autograd/profiler.py

```diff
@@ -505,12 +543,27 @@ def _privateuse1_memory_usage(mem_record):
                             f_evt.device_index,
                             f_evt.time_range.end - f_evt.time_range.start,
                         )
+                    elif f_evt.device_type == DeviceType.XPU:
```

Comment:
Why do we need a new list for XPU kernels? Can't we just reuse the kernels list in case the device type is CUDA or XPU?

Reply:
Unfortunately we can't because the single kernels list must be used for CUDA, XPU and PrivateUse1 three backends, but current logic on `use_device` can only distinguish two paths. I have explained this conflict previously in this comment https://github.com/pytorch/pytorch/pull/120185#discussion_r1497162676. And @aaronenyeshi has decided to maintain the current semantics of `use_device`. If you still have concerns or points that you believe need further discussion, please raise some hints for solving this issue while keeping current `use_device` semantics. Thank you.